nipoppy_cli/nipoppy/workflows/dataset_init.py

```diff
@@ -85,13 +121,99 @@ def run_main(self):
             FPATH_SAMPLE_MANIFEST, self.layout.fpath_manifest, log_level=logging.DEBUG
         )
 
+        if self.bids_source is not None:
+            self._init_manifest_from_bids_dataset()
+
+        if self.use_dalatad:
+            api.save(
+                dataset=self.dpath_root,
+                path=self.dpath_root / ".",
+                message="Nipoppy layout initialized.",
+            )
+
         # inform user to edit the sample files
         self.logger.warning(
             f"Sample config and manifest files copied to {self.layout.fpath_config}"
             f" and {self.layout.fpath_manifest} respectively. They should be edited"
             " to match your dataset"
         )
 
+    def _init_as_datalad_dataset(self) -> None:
+        if self.use_dalatad:
+            from datalad import api
+
+            api.create(path=self.dpath_root, result_renderer="disabled")
+            self._make_git_attributes()
+            self._make_git_ignore()
+
+    def _init_manifest_from_bids_dataset(self) -> None:
+        """Assume a BIDS dataset with session level folders.
+
+        No BIDS validation is done.
+        """
```

Comment:
This is a very rough way to get the job done. But I do not think it should be nipoppy's job to validate BIDS datasets.